torch/_higher_order_ops/invoke_subgraph.py

```diff
@@ -164,12 +173,12 @@ def backward(ctx, *grad_outs):
         contiguous_grad_outs = tuple([o.contiguous() for o in grad_outs])
```

Comment:
Yes, thats correct.

The only way to avoid the problem is to call `aot_autograd` directly for `invoke_subgraph` in like some pre-dispatch key so that you can share the same solution. But, all of us decided that calling out the dispatch keys at the pre-dispatch key feels icky.

Reply:
sounds good, just wanted to confirm / call it out. The other option is to not trace the joint graph eagerly in the autograd key, and instead have the autograd key register a node that then the the joint graph

Reply:
hmm one thing I'm a bit worried about is that today we have assertions that run in the backward, that error loudly if we guessed tangent subclass metadata incorrectly (and also give subclasses a hook to interpose and try to "fix" the bad guess.

If we are going to also potentially guess subclass tangents incorrectly at the `invoke_subgraph` backward boundaries, we risk silent incorrectness, since those error checks and hooks will not run.

Reply:
I agree this is a problem, the problem keeps getting worse lol

Reply:
Richard and I synced offline: we agreed that the silent correctness is bad, and we should either:

(1) figure out how to get the error checks

(2) avoid the possibility of guessing wrong by delaying the backward of `invoke_subgraph` until its backward is actually traced by AOTAutograd

Doing 1 actually seems quite hard, so we agreed that 2 will probably be easier to implement (and a better long term state anyway). wdyt @anijain2305 ?